packages/ui/src/ui/libs/monaco-yql-languages/helpers/getDirectoryContent.ts

```diff
@@ -0,0 +1,72 @@
+import {YTApiId, ytApiV3, ytApiV3Id} from '../../../rum/rum-wrap-api';
+import {IRange, Position, editor, languages} from 'monaco-editor';
+import ypath from '../../../common/thor/ypath';
+import {YPathItem} from '../../../pages/query-tracker/Plan/services/tables';
+import {getPathFromMonacoModel} from './getPathFromMonacoModel';
+import {QueryEngine} from '../../../pages/query-tracker/module/engines';
+import {SuggestionsWeight} from './generateSuggestions';
+import {suggestionIndexToWeight} from './suggestionIndexToWeight';
+import {getClusterProxy} from '../../../store/selectors/global';
+import YT from '../../../config/yt-config';
+
+type Props = (data: {
+    model: editor.ITextModel;
+    monacoCursorPosition: Position;
+    engine: QueryEngine;
+    range: IRange;
+}) => Promise<languages.CompletionItem[]>;
+
+export const getDirectoryContent: Props = async ({model, monacoCursorPosition, engine, range}) => {
+    const {path, cluster} = getPathFromMonacoModel(model, monacoCursorPosition, engine);
+    if (!path) return [];
+
+    try {
+        let clearPath = path.replace(/(\/$)/g, '');
+        let pathExist = await ytApiV3.exists({path: clearPath});
+        if (!pathExist) {
+            if (/[^/]\/$/.test(path)) return []; // we exclude the situation when we have the wrong path
+            clearPath = clearPath.split('/').slice(0, -1).join('/');
+            pathExist = await ytApiV3.exists({path: clearPath});
+            if (!pathExist) return [];
+        }
+
+        const clusterConfig = cluster && cluster in YT.clusters ? YT.clusters[cluster] : null;
```

Comment:
`cluster` should be always defined as a correct value else the function should not be called.
If `cluster` is not mentioned then a query is started on QT-cluster instead of current one, isn't it?

Reply:
if the cluster is not specified in the request body, then we use the structure of the current (main)cluster

Reply:
returning an empty sagest if there is no cluster